build.gradle

```diff
@@ -1,18 +1,21 @@
-plugins {
```

Comment:
I think you may run into issues when it comes to publishing if your root project is itself a "Java" project. 

You can factor out some commonality between the subprojects (they are all java projects and all have several of the same "things").